runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp

```diff
@@ -542,25 +581,45 @@ size_t J9::ARM64::JNILinkage::buildJNIArgs(TR::Node *callNode, TR::RegisterDepen
                {
                // numIntegerArgs >= properties.getNumIntArgRegs()
                int offsetInc;
+#if defined(LINUX)
+               offsetInc = 8; // always 8-byte aligned
+               op = (childType == TR::Address || childType == TR::Int64) ?
+                    TR::InstOpCode::strimmx : TR::InstOpCode::strimmw;
+#elif defined(OSX)
                if (childType == TR::Address || childType == TR::Int64)
                   {
                   op = TR::InstOpCode::strimmx;
                   offsetInc = 8;
-#if defined(OSX)
                   argOffset = (argOffset + 7) & ~7; // adjust to 8-byte boundary
-#endif
                   }
                else
                   {
-                  op = TR::InstOpCode::strimmw;
-#if defined(LINUX)
-                  offsetInc = 8;
-#elif defined(OSX)
-                  offsetInc = 4;
+                  TR::DataType nativeType = OMR::Symbol::convertSigCharToType(*sigCursor);
+                  if (nativeType == TR::Int8)
+                     {
+                     op = TR::InstOpCode::strbimm;
+                     offsetInc = 1;
+                     }
+                  else if (nativeType == TR::Int16)
+                     {
+                     op = TR::InstOpCode::strhimm;
+                     offsetInc = 2;
+                     argOffset = (argOffset + 1) & ~1; // adjust to 2-byte boundary
+                     }
+                  else if (nativeType == TR::Int32)
+                     {
+                     op = TR::InstOpCode::strimmw;
+                     offsetInc = 4;
+                     argOffset = (argOffset + 3) & ~3; // adjust to 4-byte boundary
+                     }
+                  else
+                     {
+                     TR_ASSERT(0, "Unexpected type %c", *argType);
+                     }
+                  }
```

Comment:
Same comment as above on simplifying this if/else ladder.  The one wrinkle in this code is that you do need to set `op` to a particular store instruction, but I think a small if/else or switch could accomplish that.

Reply:
I updated the code in the same way as the first loop plus a `switch` statement.